crates/fuel-core/src/executor.rs

```diff
@@ -430,12 +437,19 @@ where
         }
         block.transactions.append(&mut filtered_transactions);
 
-        coinbase_tx = self.check_coinbase(
-            block_height,
-            coinbase_tx,
-            Some(execution_data.coinbase),
-        )?;
-        self.apply_coinbase(coinbase_tx, block, execution_data, block_db_transaction)?;
+        if execution_kind != ExecutionKind::DryRun {
+            coinbase_tx = self.check_coinbase(
```

Comment:
Do we still need to do this check in production mode?

Reply:
We need only to apply coinbase, but not to check. But while the `check_coinbase` function does simple check, I think it is better to keep it to be sure that everything is okay(and makes the code consistance)=)